bitriseclient/bitriseclient.go

```diff
@@ -0,0 +1,237 @@
+	ProjectType string `json:"project_type"`
+	Provider    string `json:"provider"`
+	RepoOwner   string `json:"repo_owner"`
+	RepoURL     string `json:"repo_url"`
+	RepoSlug    string `json:"repo_slug"`
+	IsDisabled  bool   `json:"is_disabled"`
+	Status      int    `json:"status"`
+	IsPublic    bool   `json:"is_public"`
+	Owner       Owner  `json:"owner"`
+}
+
+// FetchMyAppsResponse ...
+type FetchMyAppsResponse struct {
+	Data   []Application `json:"data"`
+	Paging Paging        `json:"paging"`
+}
+
+// BitriseClient ...
+type BitriseClient struct {
+	accessToken     string
+	selectedAppSlug string
+	headers         map[string]string
+}
+
+// NewBitriseClient ...
+func NewBitriseClient(accessToken string) (client *BitriseClient, apps []Application, err error) {
+	client = &BitriseClient{accessToken, "", map[string]string{"Authorization": "token " + accessToken}}
+
+	log.Infof("Asking your application list from Bitrise...")
+
+	requestURL, cerr := urlutil.Join(baseURL, appsEndPoint)
+	if cerr != nil {
+		err = cerr
+		return
+	}
+
+	log.Debugf("\nRequest URL: %s", requestURL)
+
+	// Response struct
+	var appListResponse FetchMyAppsResponse
+	responseStatusCode := -1
+
+	stillPaging := true
+	var next string
+	for stillPaging {
+
+		headers := client.headers
+
+		request, cerr := createRequest(http.MethodGet, requestURL, headers, nil)
+		if cerr != nil {
+			err = cerr
+			return
+		}
+
+		if len(next) > 0 {
+			quearryValues := request.URL.Query()
+			quearryValues.Add("next", next)
+			request.URL.RawQuery = quearryValues.Encode()
+		}
+
+		//
+		// Perform request
+		if cerr := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+			body, statusCode, err := performRequest(request)
+			if err != nil {
+				log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+				if !strings.Contains(err.Error(), "failed to perform request") {
+					log.Warnf("Response status: %d", statusCode)
+					log.Warnf("Body: %s", string(body))
+				}
+				return err
+			}
+
+			responseStatusCode = statusCode
+
+			// Parse JSON body
+			if err := json.Unmarshal([]byte(body), &appListResponse); err != nil {
+				return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+			}
+			return nil
+
+		}); cerr != nil {
+			err = cerr
+			return
+		}
+
+		logDebugPretty(appListResponse)
+
+		apps = append(apps, appListResponse.Data...)
+
+		if len(appListResponse.Paging.Next) > 0 {
+			next = appListResponse.Paging.Next
+			appListResponse = FetchMyAppsResponse{}
+		} else {
+			stillPaging = false
+		}
+
+	}
+
+	return
+}
+
+// SetSelectedAppSlug ...
+func (client *BitriseClient) SetSelectedAppSlug(slug string) {
+	client.selectedAppSlug = slug
+}
+
+func createUploadRequest(requestMethod string, url string, headers map[string]string, filePth string) (*http.Request, error) {
+	var fContent []byte
+
+	f, err := os.Open(filePth)
+	if err != nil {
+		return nil, err
+
+	}
+
+	fContent, err = ioutil.ReadAll(f)
```

Comment:
fContent, err :=


we do not use fContent before this line, also `content` is enough for var name